docs/cloud-tutorials/best-practices.md

```diff
@@ -0,0 +1,87 @@
+
+### Swapping database instance
+While an application cannot change database once registered, you can change the database instance (server) hosting the application database. When doing so, you can also decide to redeploy a previous version of the application. This is how DBOS Cloud support point in time recovery today.
+
+## Application databases
+
+:::tip
+Application databases are "logical" postgres databases. They are not to be confused with database *instances*, the server hosting application databases.
+:::
+
+* Application databases cannot be shared across applications
+* An application cannot change its database once deployed
+
+### Database roles
+
+* DBOS Cloud will create the following roles for each application:
+    * `dbosadmin` - a privileged role used to execute your schema migration steps during application deployment
+    * `approle` - a user-provided role you provide when provisioning the database instance. This role has
+        * a CREATE grant on the application database
+        * `pg_read_all_data` and `pg_write_all_data` grants
+        * an ALL PRIVILEGES grant on all schemas and [base tables](https://www.postgresql.org/docs/current/infoschema-tables.html#INFOSCHEMA-TABLES) in the database instance, expect for the schemas `pg_catalog`, `information_schema`, `dbos` or `pg_toast` schemas
+        * ownership on all base tables in the database instance
+    * `dbosproxyrole` - a read-only role used during [DBOS Time Travel](cloud-tutorials/timetravel-debugging)
+
+### Schema migrations
+* You can instruct DBOS Cloud to run [migration commands](api-reference/configuration#database) during application deploy. These are handled by `dbosadmin`. You can use them to install extensions, create tables, etc.
+* For more complex operations, you can use the DBOS Cloud CLI to connect to your database and run migrations manually.
+
+### System databases
+* DBOS Transact relies on a _system database_ to store metadata about the application. This database cannot be configured when running on DBOS Cloud.
+
+## Database instances
+
+### Provisioning instances
+* Provisioning instances is an asynchronous process. You can check the status of the instance with `npx dbos-cloud db status <database-instance-name>`
+
+### Linked databases (BYODB)
+* DBOS Cloud will not configure `approle` for linked database
+* DBOS Cloud will not provide a default SSL certificate for linked database [FIXME: what is the expected flow here?]
+
+## Users and organizations
+
+### Registering a new user
+- A new user needs to be signed up with DBOS Cloud (through auth0) before registering a Cloud account.
+
+### Refresh tokens
+The preferred way to programmatically authenticate with DBOS Cloud is to use refresh tokens. Note they:
+* Expire after a year or a month of inactivity
+* Can be revoked manually
+
+## Time travel
+* Time travel will result in the creation of a new application database in the database instance
```

Comment:
```suggestion
* Time travel will result in the creation of a new database called `<app_db_name>_dbos_prov` in the database server. 
```
Provide a link to: https://docs.dbos.dev/api-reference/system-tables#provenance-tables